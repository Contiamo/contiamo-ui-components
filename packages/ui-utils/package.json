{
  "name": "contiamo-ui-utils",
  "version": "0.0.1-44",
  "description": "A collection of utilities used within the Contiamo UI suite",
  "main": "./lib/index.js",
  "typings": "./lib/index.d.ts",
  "repository": "git@github.com:Contiamo/ui-components.git",
  "author": "Tejas Kumar <tejas@contiamo.com>",
  "license": "MIT",
  "scripts": {
    "package": "rm -rf lib && tsc -d --outDir './lib'",
    "package:watch": "rm -rf lib && tsc -d -w",
    "test": "jest",
    "test:watch": "jest --watch",
    "lint": "tslint './src/**/*.{ts,tsx}' --fix",
    "prettier": "prettier './src/**/*.{ts,tsx}' --write",
    "precommit": "npm run test && npm run lint && npm run prettier && npm run package && git add .",
    "preversion": "npm run precommit",
    "version": "npm run package && git add ."
  },
  "dependencies": {
    "contiamo-ui-theme": "^0.0.1-44",
    "glamor": "^2.20.40",
    "glamorous": "^4.9.7",
    "react": "^16.0.0",
    "tinycolor2": "^1.4.1"
  }
}
