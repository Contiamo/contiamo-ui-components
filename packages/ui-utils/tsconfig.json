{
  "compilerOptions": {
    "outDir": "lib",
    "target": "ES5",
    "module": "commonjs",
    "sourceMap": true,
    "noImplicitAny": true,
    "jsx": "react",
    "lib": ["ES6", "DOM", "DOM.Iterable", "ScriptHost"],
    "moduleResolution": "node"
  },
  "include": ["./src/index.ts", "./types/**/*"],
  "exclude": ["./node_modules"]
}
