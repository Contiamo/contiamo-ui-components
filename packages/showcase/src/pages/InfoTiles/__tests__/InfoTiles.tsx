import * as React from "react"
import { render } from "enzyme"

import { wrapTheme } from "@contiamo/ui-utils"
import { contiamoTheme } from "@contiamo/ui"
import ThemelessInfoTiles from "../InfoTiles"

const InfoTiles = wrapTheme(contiamoTheme)(ThemelessInfoTiles)

describe("InfoTiles Page", () => {
  it("Should render correctly", () => {
    expect(render(<InfoTiles />)).toMatchSnapshot()
  })
})
