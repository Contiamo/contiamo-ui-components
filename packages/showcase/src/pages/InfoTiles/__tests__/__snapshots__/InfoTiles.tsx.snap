// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards Page Should render correctly 1`] = `
<div>
  <div
    class="css-1kznpaj"
  >
    InfoTiles
  </div>
  <p>
    InfoTiles are little pieces of information that contain a descriptor and a value of the descriptor. These elements are commonly used to represent KPI data, with strong key-value pairings.
  </p>
  <div
    style="display:flex;"
  >
    <div
      class="css-farlai"
    >
      <small
        class="css-p41l92"
      >
        Components
      </small>
      <span>
        200,000
      </span>
    </div>
    <div
      class="css-farlai"
    >
      <small
        class="css-p41l92"
      >
        Reusability
      </small>
      <span>
        High
      </span>
    </div>
    <div
      class="css-farlai"
    >
      <small
        class="css-p41l92"
      >
        Potential Added Value
      </small>
      <span>
        â‚¬2M
      </span>
    </div>
  </div>
  <h4>
    Usage
  </h4>
  <div
    class="playground"
  >
    <div
      class="playgroundCode"
    >
      <div
        class="ReactCodeMirror playgroundStage"
      >
        <textarea
          autocomplete="off"
        >
          [object Object]
        </textarea>
      </div>
    </div>
    <div
      class="playgroundPreview"
    >
      <div>
        <div
          class="previewArea"
        />
      </div>
    </div>
  </div>
  <h4>
    Props
  </h4>
  <table
    class="css-xa6ngs"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Description
        </th>
        <th>
          Default
        </th>
        <th>
          Type
        </th>
        <th>
          Optional?
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          label
        </td>
        <td>
          What is the key in the key-value pairing? This is the description of the statistic itself.
        </td>
        <td />
        <td>
          string
        </td>
        <td>
          ðŸš«
        </td>
      </tr>
      <tr>
        <td>
          color
        </td>
        <td>
          See above. A stat can have its own background color. This can be a hex code, or a named color from your theme.
        </td>
        <td>
          white
        </td>
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          icon
        </td>
        <td>
          React Feather icon name. See \`Icon\` component.
        </td>
        <td />
        <td>
          string
        </td>
        <td>
          âœ…
        </td>
      </tr>
      <tr>
        <td>
          onIconClick
        </td>
        <td>
          Method triggered when the icon is clicked.
        </td>
        <td />
        <td>
          function
        </td>
        <td>
          âœ…
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
